vault/plugin_catalog.go

```diff
@@ -348,20 +348,14 @@ func (c *PluginCatalog) newPluginClient(ctx context.Context, pluginRunner *plugi
 
 	clientConn := rpcClient.(*plugin.GRPCClient).Conn
 
-	muxed, err := pluginutil.MultiplexingSupported(ctx, clientConn)
+	muxed, err := pluginutil.MultiplexingSupported(ctx, clientConn, config.Name)
 	if err != nil {
 		return nil, err
 	}
 
-	if muxed {
-		// Wrap rpcClient with our implementation so that we can inject the
-		// ID into the context
-		pc.clientConn = &pluginClientConn{
-			ClientConn: clientConn,
-			id:         id,
-		}
-	} else {
-		pc.clientConn = clientConn
+	pc.clientConn = &pluginClientConn{
```

Comment:
Why are we no longer using the value of `muxed` to conditionally wrap the RPC Client? Looks like we aren't using the returned boolean and are always enabling multiplexing by wrapping the RPC Client with `pluginClientConn`

Reply:
The ID is necessary to serve a mplexed plugin in "splexed mode", otherwise we encounter errors such as:
```
* error creating database object: invalid database version: 2 errors occurred:
* error getting plugin type: unable to get database plugin type: rpc error: code = Unknown desc = unexpected number of IDs in metadata: (0)
* Incompatible API version with plugin. Plugin version: 5, Client versions: [3 4]
```
Having an id is expected to manage a mplex-compatible plugin instance regardless of what mode it operates in.

The reasoning to always use a mplex-compatible client is:
* Since there is an explicit flag and utility method responsible to determine if mplexed is supported or not (`extPlugin.multiplexingSupport = muxed, pluginutil.MultiplexingSupported`), having an id even if unused doesn't cause ambiguity
* Since in the long run we want mplex to be the default and possibly only mode available, removing a condition brings us closer to the desired goal rather than adding a lot of logic to handle if we're dealing with the wrapped client or not

Reply:
Got it, thanks for the detailed explanation :)